portability-spi-cloud/src/main/java/org/dataportabilityproject/spi/cloud/types/PortabilityJob.java

```diff
@@ -84,4 +89,52 @@ public LocalDateTime getLastUpdateTimestamp() {
     public void setLastUpdateTimestamp(LocalDateTime lastUpdateTimestamp) {
         this.lastUpdateTimestamp = lastUpdateTimestamp;
     }
+
+    public JobAuthorization getJobAuthorization() {
+        return jobAuthorization;
+    }
+
+    /**
+     * Sets and validates the {@link JobAuthorization} data associated with this job.
+     */
+    public void setJobAuthorization(JobAuthorization jobAuthorization) {
+        switch(jobAuthorization.getState()) {
+            case INITIAL:
+                validateState(jobAuthorization, true, false, false, false, false);
+                break;
+            case PENDING_WORKER_ASSIGNMENT:
+                validateState(jobAuthorization, true, false, false, false, false);
+                break;
+            case ASSIGNED_WITHOUT_AUTH_DATA:
+                validateState(jobAuthorization, true, false, false, true, true);
+                break;
+            case ASSIGNED_WITH_AUTH_DATA:
+                validateState(jobAuthorization, true, true, true, true, true);
+                break;
+        }
+        this.jobAuthorization = jobAuthorization;
+    }
+
+    /**
+     * Validate the correct data is present for each state transition.
+     */
+    private void validateState(JobAuthorization jobAuthorization,
+        boolean sessionKeyExists,
```

Comment:
this makes me nervous that the wrong flags will get passed in.. can you make less error prone?

Reply:
How about this?